README.adoc

```diff
@@ -36,30 +38,238 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
+
+```
+<dependencies>
+    ...
+    <dependency>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-cloud-gcp-starter-pubsub</artifactId>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+        <groupId>org.springframework.cloud</groupId>
+        <artifactId>spring-integration-gcp</artifactId>
+        <version>1.0.0.BUILD-SNAPSHOT</version>
+    </dependency>
+    ...
+</dependencies>
+
+
+
+<repositories>
+    ...
+    <repository>
+        <id>repository.spring.snapshot</id>
+        <name>Spring Snapshot Repository</name>
+        <url>http://repo.spring.io/snapshot</url>
+    </repository>
+    ...
+</repositories>
+```
+
+Or, if you're using Gradle:
+
+```
+repositories {
+    ...
+    maven {
+        url "http://repo.spring.io/snapshot"
+    }
+    ...
+}
+
+dependencies {
+    ...
+    compile("org.springframework.cloud:spring-cloud-gcp-starter-pubsub:1.0.0.BUILD-SNAPSHOT")
+    compile("org.springframework.cloud:spring-integration-gcp:1.0.0.BUILD-SNAPSHOT")
+    ...
+}
+```
+
+== Create application files
+
+We'll need a class to include our channel adapter and messaging configuration. Create a
+PubSubApplication class with the @SpringBootApplication header, as is typical with a Spring Boot
+application.
+
+`src/main/java/hello/PubSubApplication.java`
+[source, java]
+----
+@SpringBootApplication
+public class PubSubApplication {
+
+  public static void main(String[] args) throws IOException {
+    SpringApplication.run(PubSubApplication.class, args);
+  }
+
+}
+----
+
+Additionally, since we're building a web application, you can also create a WebAppController class
+to separate between the controller and configuration logic, although they can also live in the same
+place.
+
+`src/main/java/hello/WebAppController.java`
+[source, java]
+----
+@RestController
+public class WebAppController {
+}
+----
+
+We're still missing two files: the HTML and the properties one.
+
+`src/main/resources/static/index.html`
+[source, java]
+----
+include::/complete/src/main/resources/static/index.html[]
+----
+
+`src/main/resources/application.properties`
+[source, java]
+----
+include::/complete/src/main/resources/application.properties[]
+----
 
-Create a new controller for your Spring application:
+The Spring Cloud GCP Core Boot starter can auto-configure these two properties and make them
+optional. Although if the properties are specified here, they overwrite the auto-configured values.
+The Spring Cloud GCP Core Boot starter is bundled with the Spring Cloud GCP Pub/Sub Boot starter.
 
-`src/main/java/hello/GreetingController.java`
+The GCP project ID is auto-configured from the GOOGLE_CLOUD_PROJECT environment variable, among
+https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L287[several other sources].
+The OAuth2 credentials are auto-configured from the
```

Comment:
To be honest, for a starter guide, having to go through the trouble of creating a service account just to be able to use it is too much. Also, that link suggests users to change the `GOOGLE_APPLICATION_CREDENTIALS` which, although correct, should be done differently in Spring (using the `spring.cloud.gcp.credentialsLocation` property). Or using `gcloud auth` to login with a user or service account, in which case the environment variable is automatically filled.